src/react/GroupDevicesPage.tsx

```diff
@@ -18,23 +18,83 @@
 
 import React, { useEffect, useState, useCallback } from 'react';
 import { Button, Modal, OverlayTrigger, Spinner, Table, Tooltip } from 'react-bootstrap';
+import AstarteClient, { AstarteDevice } from 'astarte-client';
 
 import { Link } from 'react-router-dom';
 import SingleCardPage from './ui/SingleCardPage';
 
-const deviceTableRow = (device, index, showModal) => {
+const CircleIcon = React.forwardRef<HTMLElement, React.HTMLProps<HTMLElement>>((props, ref) => (
```

Comment:
I am following linter's rules about not using functions before their definition. The rule is of course questionable, I am open for discussion

Reply:
I remember the rule, but the linter is not complaining before, so i was wondering why it would start complaining now

Reply:
Because the rule is enforced for Typescript code. We can decide whether this is fine or not. We also could leave the rule active for variables but not for functions (declared with `function f() {}`)

Reply:
if typescript enforces it, there must be a valid reason behind. Let's stick with it